src/ceph-volume/ceph_volume/devices/lvm/strategies/strategies.py

```diff
@@ -2,13 +2,21 @@
 
 class Strategy(object):
 
-    def __init__(self, devices, args):
+    def __init__(self, data_devs, db_devs, wal_devs, args):
```

Comment:
now that all strategies are inheriting from this, it makes it odd to have the same parent for both filestore and bluestore. Filestore doesn't have a db_dev or a wal_dev but somehow the child classes for filestore use this forcing the code to have to pick either db_devs or wal_devs. 

Either these parent classes need to be split into bluestore and filestore or the whole sub-classing would have to go

Reply:
Python does not allow for ctor overloading, so I went with the most explicit version (which happens to be bluestore on three devices).

Splitting this into bluestore and filestore specific implementations will again add duplicate code. I took the pragmatic route. I think this can be resolved with better naming and a comment, like
``` python
def __init__(self, data_devs, db_or_journal_devs, wal_devs, args):
'''
Note that any filestore strategy will always pass an empty list for wal_devs
'''
```

Not sure what can be gained by going back on the subclassing?

Reply:
The proposed implementation is trying to save a few lines of code but adding incorrect definitions for devices (journals aren't present in filestore, and it is using db_devs which don't exist)

The gain here would be that it would be clear that filestore uses journals, not db. At the cost of some duplication. I would prefer clarity over Lines Of Code savings

Reply:
Just to be sure: I changed the name prior to your comment.

The saved lines of code actually stem from #25238. Not sure I'd call ~50 loc "a few lines". In any case the main goal was to remove duplicate implementations in four strategy constructors to adhere a bit more to a DRY approach.

I'm certainly happy to add some boilerplate code in order to make a name in an internal implementation less confusing.